### PR TITLE
Color contrast

### PR DESCRIPTION
**Pull request by allcaps**
This PR increases color contrast in the Wagtail admin.
Chrome Lighthouse is used to check for accessibility errors on contrast. All contrast tests pass.
I used the design by @benenright 
unnamed

The individual commits might give additional insight in my work process.
- Increase text on white contrast
- Detach nav colors from color-grey vars gives fine grained color control
- Simplify color functions and round numbers.
- Remove color from privacy icon to make hover state more notable and design less complex
- Stop style bleeding on listing. Make Privacy text label white
- Increase breadcrumb contrast
- Drop redundant color color-grey-1-1
- Use color variable
- Use color functions where there is a obvious relation between them

Current:
simulator screen shot - ipad air - 2018-09-26 at 23 58 43

New:
simulator screen shot - ipad air - 2018-09-26 at 23 58 21

Also note the changes to search input and privacy toggle and the body text.
screen_shot_2018-09-26_at_11_25_39

Checks:
* Do the tests still pass? YES
* Does the code comply with the style guide? YES
* For Python changes: N/A
* For front-end changes. Tested browsers:

- Mobile Safari | iOS Phone => iPhone 6s IOS 9 Safari, iPhone XS Max 12 Safari
- Mobile Safari | iOS Tablet  => iPad Air2 IOS 8 Safari, iPad Air 12 Safari
- Chrome | Android => Google Nexus 6 Chrome 63 && Samsung Galaxy S5 Android 4.4 Chrome 
- IE | Desktop | 11 => Win7 IE11
- Chrome | Desktop | Last 2
- MS Edge | Desktop => Windows 10 Edge 17
- Firefox | Desktop => Windows 10 Firefox 61 and 62
- Firefox ESR | Desktop | OSX 10.13.6 Firfox ESR 60
- Safari | macOS => OSX 10.13.6 Safari 12

* For new features: N/A

### CODE REVIEW

File: wagtail/admin/static_src/wagtailadmin/scss/components/_listing.scss [hunk 1]
```diff
@@ -377,7 +375,7 @@ ul.listing {
         .button {
             background-color: $color-white;
             color: $color-teal;
-            border-color: $color-teal;
+            border-color: rgba(0, 0, 0, 0.25);
```
Comment (nmorduch) [inline wagtail/admin/static_src/wagtailadmin/scss/components/_listing.scss]: I'm finding this to be low enough contrast that I don't reliably read these as being buttons. I played with it a tiny bit in the browser and these options seemed ok: `$color-teal-darker` `$color-teal-dark` `rgba(0, 0, 0, 0.4)`

File: wagtail/admin/static_src/wagtailadmin/scss/_variables.scss [hunk 2]
```diff
@@ -25,9 +25,10 @@ $breakpoint-desktop-large: 75em; // 1200px
 $breakpoint-desktop-larger: 100em; // 1600px
 
 // colours
-$color-teal: #43b1b0;
-$color-teal-darker: darken($color-teal, 10%);
-$color-teal-dark: #246060;
+$color-teal: #007d7e;
+$color-teal-darker: darken(adjust-hue($color-teal, 1), 3);
```
Comment (nmorduch) [inline wagtail/admin/static_src/wagtailadmin/scss/_variables.scss]: I'm very much in favor of the change to using color functions! My personal preference would be to add a code comment reminding that if the base colors change in the future the functions may also need adjusting, but that's far from necessary.

File: docs/_static/css/custom.css [hunk 3]
```diff
@@ -1,7 +1,7 @@
 body.wy-body-for-nav {
     font-family: -apple-system, BlinkMacSystemFont, 'Helvetica Neue', Helvetica, Arial, sans-serif;
     line-height: 1.5em;
-    color: #333; /*$color-grey-1-1*/
+    color: #333;
```
[the remark this answers is not in the document]
Comment (allcaps) [reply docs/_static/css/custom.css]: First of all, thanks for the review! I really appreciate your effort.

I think not. `docs/_static/css/custom.css` is a documentation stylesheet and is not in the preprocessed css workflow at all. I wonder what the original author intended. If you update the value of `$color-grey-1-1` update this hardcoded value as well? The thing is, _this is the only place_ where `$color-grey-1-1` is used AND it is not really 'used' it's only a comment.

So bottom line, there is a reference to a Wagtail interface color, but that color isn't in the Wagtail interface at all. So the comment is redundant.

I'm happy to fix the documentation colors to match the wagtail interface. But like to keep that out of this PR. As it is a little more work to make an review already.

File: wagtail/admin/static_src/wagtailadmin/scss/components/_listing.scss (hunk 1 above)
Comment (allcaps) [reply wagtail/admin/static_src/wagtailadmin/scss/components/_listing.scss]: I'll have another look at it. Buttons are defined in multiple places and in various ways. The used `0.25` as it matches the 'more' button outline.

Comment (nmorduch) [reply wagtail/admin/static_src/wagtailadmin/scss/components/_listing.scss]: Oh wow, now that I look at it you're right that the buttons are entirely inconsistent. I'd love to make a PR to fix that. I do still think that darkening the button borders is important, and I think it would make more sense for that to be grouped with color changes than button style consistency. But I'm ok to do it there if you'd rather not do it here.

For reference, now that I have this locally, here's `rgba(0, 0, 0, 0.35)` which I think is a pretty minimal change but makes me feel better:
image
vs the current `rgba(0, 0, 0, 0.25)`:
image
and the current on master:
image

File: wagtail/admin/static_src/wagtailadmin/scss/components/_tabs.scss [hunk 4]
```diff
@@ -18,7 +18,7 @@
 
     a {
         @include transition(border-color 0.2s ease);
-        background-color: lighten($color-teal-darker, 3%);
+        background-color: $color-teal-darker;
```
Comment (nmorduch) [inline wagtail/admin/static_src/wagtailadmin/scss/components/_tabs.scss]: I'm unfortunately also finding this contrast to be insufficient, and finding the hover style to be be almost unnoticeable with the new colors. I'm not happy with any of the color options I'm thinking of at the moment, but it's also getting late here, so I'm hoping to have fresher eyes in the daytime.

This branch:
image
master:
image

The middle tab has `:hover` applied on both

File: wagtail/admin/static_src/wagtailadmin/scss/_variables.scss [hunk 5]
```diff
@@ -38,22 +39,24 @@ $color-white: #fff;
 $color-black: #000;
 
 // darker to lighter
-$color-grey-1: #333;
-$color-grey-1-1: #404040;
-$color-grey-2: #666;
-$color-grey-3: #d9d9d9;
-$color-grey-4: #e6e6e6;
-$color-grey-5: #fafafa;
+$color-grey-1: darken($color-white, 80);
```
[the remark this answers is not in the document]
Comment (allcaps) [reply wagtail/admin/static_src/wagtailadmin/scss/_variables.scss]: Okay, added `$color-grey-1` back in.

**Comment by mikedingjan** [comment]
LGTM 👍 

What I did to check:
  - Read all the comments in the PR and checked their related commits
  - Did a checkout of the PR and ran it locally, clicked around to see if nothing breaks
  - Was amazed by how much better I like the new contrast!

Great work!

File: wagtail/admin/static_src/wagtailadmin/scss/components/_tabs.scss (hunk 4 above)
Comment (allcaps) [reply wagtail/admin/static_src/wagtailadmin/scss/components/_tabs.scss]: @nmorduch I think this is better. Do you agree?

**Comment by allcaps** [comment]
@nmorduch Will you bee so kind to review the last three commits and approve? I think this PR is ready to be merged. I'd like your approval.

**Comment by gasman** [comment]
For the purpose of the release candidate I'll take @nmorduch's comment as permission to go ahead and merge :-) No problem to roll back for the final release if need be.

File: wagtail/admin/static_src/wagtailadmin/scss/components/_tabs.scss (hunk 4 above)
Comment (nmorduch) [reply wagtail/admin/static_src/wagtailadmin/scss/components/_tabs.scss]: Yes, I think this is much better!
